### ISSUE TITLE
Look into message queue options

### ISSUE THREAD

**Issue by smarlowucf** [SUSE-Enceladus/mqsf#2]
Mash relies on AMQP. Consider the advantages/disadvantages of this and also MQTT and ZeroMQ. Consider what is currently leveraged from AMQP for Mash. Do we need these features? Is it worth the overhead of a broker? Can it be replaced by native code in conjunction with ZeroMQ?

**Comment by apozsuse** [comment]
I've been reading a bit about the different architectures ZeroMQ enables being brokerless in this paper [this paper](http://wiki.zeromq.org/whitepapers:brokerless).
Not sure if the amount of messages we'll have to handle justifies implementing a directory service vs using a message broker (like mash uses AMQP for now).

**Comment by smarlowucf** [comment]
Yeah, I think there's some good reasons to stick to amqp or mqtt. It does add a lot of overhead to have a broker running but it addes features that we would need to implement if we went with brokerless zeromq.
